File: backend-services/projectservice/src/main/java/com/mariusniemet/projectservice/resolvers/LabelsResolver.java
package com.mariusniemet.projectservice.resolvers;

import com.mariusniemet.projectservice.dto.CreateLabelInput;
import com.mariusniemet.projectservice.dto.UpdateLabelInput;
import com.mariusniemet.projectservice.models.Label;
import com.mariusniemet.projectservice.models.Project;
import com.mariusniemet.projectservice.services.LabelsService;
import com.netflix.graphql.dgs.*;

import java.util.List;
import java.util.Map;

@DgsComponent
public class LabelsResolver {
    private LabelsService service;

    public LabelsResolver(LabelsService service) {
        this.service = service;
    }

    @DgsQuery
    public List<Label> labels(@InputArgument Integer project_id, @InputArgument Integer team_id){
        System.out.println(team_id);
        return this.service.findAll(project_id, null);
    }

    @DgsQuery
    public Label label(@InputArgument Integer id){
        return this.service.findOne(id);
    }

    @DgsMutation
    public Label createLabel(@InputArgument CreateLabelInput createLabelInput){
        return this.service.create(createLabelInput);
    }

    @DgsMutation
    public Label updateLabel(@InputArgument UpdateLabelInput updateLabelInput){
        return this.service.update(updateLabelInput);
    }

    @DgsEntityFetcher(name = "Label")
    public Label label(Map<String, Object> values) {
        return this.service.findOne((Integer) values.get("id"));
    }
}

File: backend-services/organization/src/labels/labels.resolver.ts
import {
  Args,
  Int,
  Mutation,
  Query,
  ResolveReference,
  Resolver,
} from '@nestjs/graphql';
import { CreateLabelInput } from './dto/create-label.input';
import { UpdateLabelInput } from './dto/update-label.input';
import { Label } from './entities/label.entity';
import { LabelsService } from './labels.service';

@Resolver(() => Label)
export class LabelsResolver {
  constructor(private readonly labelsService: LabelsService) {}

  @Mutation(() => Label)
  createLabel(@Args('createLabelInput') createLabelInput: CreateLabelInput) {
    return this.labelsService.create(createLabelInput);
  }

  @Query(() => [Label], { name: 'labels' })
  findAll(
    @Args('project_id', { type: () => Int, nullable: true }) project_id: number,
    @Args('team_id', { type: () => Int, nullable: true }) team_id: number,
  ) {
    return this.labelsService.findAll(project_id, team_id);
  }

  @Query(() => Label, { name: 'label' })
  findOne(@Args('id', { type: () => Int }) id: number) {
    return this.labelsService.findOne(id);
  }

  @Mutation(() => Label)
  updateLabel(@Args('updateLabelInput') updateLabelInput: UpdateLabelInput) {
    return this.labelsService.update(updateLabelInput.id, updateLabelInput);
  }

  @Mutation(() => Label)
  removeLabel(@Args('id', { type: () => Int }) id: number) {
    return this.labelsService.remove(id);
  }

  @ResolveReference()
  resolveReference(reference: {
    __typename: string;
    id: number;
  }): Promise<Label> {
    return this.labelsService.findOne(reference.id);
  }
}
